pkg/ttl/ttlworker/job_manager.go

```diff
@@ -66,6 +66,11 @@ const taskGCTemplate = `DELETE task FROM
 const ttlJobHistoryGCTemplate = `DELETE FROM mysql.tidb_ttl_job_history WHERE create_time < CURDATE() - INTERVAL 90 DAY`
 const ttlTableStatusGCWithoutIDTemplate = `DELETE FROM mysql.tidb_ttl_table_status WHERE (current_job_status IS NULL OR current_job_owner_hb_time < %?)`
 
+// don't remove the rows for non-exist tables directly. Instead, set them to cancelled. In some special situations, the TTL job may still be able
+// to finish correctly. If that happen, the status will be updated from 'cancelled' to 'finished' in `(*ttlJob).finish`
+const ttlJobHistoryGCNonExistTableTemplate = `UPDATE mysql.tidb_ttl_job_history SET status = 'cancelled'
```

Comment:
Maybe it'd be better to not remove the `tidb_ttl_table_status` if the current state is running? And let the job owner to cancel and remove the status / update the history.

I'm not sure. This solution is simpler, but a little ugly :facepalm: .

Reply:
Yes, I think always takeover unfinished job is better. We can optimize it in future PRs.

Reply:
It's good. I'll do this change later in another PR.